### PR TITLE
update glibc to 2.35

### PR DESCRIPTION
**Pull request by bcressey**
**Issue number:**
#1976

**Description of changes:**
Updated glibc to 2.35. Refreshed patches, adjusted configure options, and added C.UTF-8 to installed files.

**Testing done:**
Verified that aarch64 and x86_64 variants built, joined a cluster, and passed conformance testing.

Confirmed that the C.UTF-8 locale was working as expected:
```
bash-5.0# touch /etc/ßash
bash-5.0# LANG=C ls -1 /etc/ßash
'/etc/'$'\303\237''ash'
bash-5.0# LANG=C.UTF-8 ls -1 /etc/ßash
/etc/ßash
```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Comment by bcressey** [comment]
For x86_64:
```
  #19 118.8 + make -j1 install_root=/home/builder/rpmbuild/BUILDROOT/bottlerocket-x86_64-glibc-2.35-1.x86_64 install-files-C.UTF-8/UTF-8 -C localedata objdir=/home/builder/rpmbuild/BUILD/glibc-2.35/build
  #19 118.8 FATAL: kernel too old
  #19 118.8 make: *** [Makefile:475: install-files-C.UTF-8/UTF-8] Error 127
```

For aarch64:
```
  #19 111.5 + make -j1 install_root=/home/builder/rpmbuild/BUILDROOT/bottlerocket-aarch64-glibc-2.35-1.x86_64 install-files-C.UTF-8/UTF-8 -C localedata objdir=/home/builder/rpmbuild/BUILD/glibc-2.35/build
  #19 111.5 /bin/sh: line 2: /home/builder/rpmbuild/BUILD/glibc-2.35/build/elf/ld-linux-aarch64.so.1: cannot execute binary file: Exec format error
  #19 111.5 make: *** [Makefile:475: install-files-C.UTF-8/UTF-8] Error 126
  ```

Pretty clear that the cross-compiling path isn't being taken, or else we'd skip these tests.

**Comment by bcressey** [comment]
Trying a slightly different invocation for installing the C.UTF-8 locale, under the theory that we're not finding `config.make` with the right `cross-compiling = yes` entry.

**Comment by bcressey** [comment]
Ugh. I promise this all works fine locally! Something strange is happening on the CI build hosts.

**Comment by bcressey** [comment]
"Something strange" is that I have a newer kernel on my development host, and also ended up with the QEMU static programs installed, which masks both these errors.

**Comment by bcressey** [comment]
Need this fix for chronyd's seccomp filter:
https://git.tuxfamily.org/chrony/chrony.git/commit/?id=8bb8f15a

**Comment by bcressey** [comment]
(rebase only)

**Comment by bcressey** [comment]
(add chronyd patch for glibc 2.3.5, add latest glibc patches)

File: packages/glibc/glibc.spec [hunk 1]
```diff
@@ -9,42 +9,46 @@ Source1: glibc-tmpfiles.conf
 Source2: ld.so.conf
 Source3: ldconfig-service.conf
 
-# Upstream patches from 2.34 release branch:
+# We include this patch as a source file to have more control over how it's
+# applied and reverted during the build.
+Source99: HACK-only-build-and-install-localedef.patch
```
Comment (arnaldo2792) [inline packages/glibc/glibc.spec:14]: ~~Nit: shouldn't this be after `000X` commits?~~

I misread the PR, `HACK-only-build-and-install-localedef.patch` isn't a patch applied by `rpm` :man_facepalming:, this looks good :+1:

**Review by arnaldo2792 — APPROVED** [review]
:tada:

**Review by etungsten — APPROVED** [review]

**Review by cbgbt — APPROVED** [review]
Nice work!

:hot_pepper:

**Pull request merged**